ci/live-cluster-sanity.sh

```diff
@@ -0,0 +1,142 @@
+
+  validator_log="$cluster_label-validator.log"
+  ./net/ssh.sh "$instance_ip" -Llocalhost:18899:localhost:18899 ./solana-validator \
+    --no-untrusted-rpc \
+    --ledger cluster-sanity/ledger \
+    --log - \
+    --init-complete-file cluster-sanity/init-completed \
+    --enable-rpc-exit \
+    --private-rpc \
+    --rpc-port 18899 \
+    --rpc-bind-address localhost \
+    --snapshot-interval-slots 0 \
+    "$@" &> "$validator_log" &
+  ssh_pid=$!
+  tail -F "$validator_log" > cluster-sanity/log-tail 2> /dev/null &
+  tail_pid=$!
+
+  attempts=100
+  while ! ./net/ssh.sh "$instance_ip" test -f cluster-sanity/init-completed &> /dev/null ; do
+    attempts=$((attempts - 1))
+    if [[ (($attempts == 0)) || ! -d "/proc/$ssh_pid" ]]; then
+      on_error "start"
+    fi
+
+    sleep 3
+    echo "##### validator is starting... (until timeout: $attempts) #####"
+    show_log
+  done
+
+  echo "--- Monitoring validator $cluster_label"
+
+  snapshot_slot=$(./net/ssh.sh "$instance_ip" ls -t cluster-sanity/ledger/snapshot* |
+    head -n 1 |
+    grep -o 'snapshot-[0-9]*-' |
+    grep -o '[0-9]*'
+  )
+  current_root=$snapshot_slot
+  goal_root=$((snapshot_slot + 100))
+
+  attempts=100
+  while [[ $current_root -le $goal_root ]]; do
+    attempts=$((attempts - 1))
+    if [[ (($attempts == 0)) || ! -d "/proc/$ssh_pid" ]]; then
+      on_error "root new slots"
+    fi
+
+    sleep 3
+    current_root=$(./target/release/solana --url http://localhost:18899 slot --commitment root)
+    echo "##### validator is running ($current_root/$goal_root)... (until timeout: $attempts) #####"
+    show_log
+  done
+
+  _ curl \
+    -X POST \
+    -H 'Content-Type: application/json' \
+    -d '{"jsonrpc":"2.0","id":1, "method":"validatorExit"}' \
+    http://localhost:18899
+
+  (sleep 3 && kill "$tail_pid") &
+  kill_pid=$!
+  wait "$ssh_pid" "$tail_pid" "$kill_pid"
+}
+
+# UPDATE docs/src/clusters.md TOO!!
+test_with_live_cluster "mainnet-beta" \
+    --entrypoint mainnet-beta.solana.com:8001 \
+    --trusted-validator 7Np41oeYqPefeNQEHSv1UDhYrehxin3NStELsSKCT4K2 \
+    --trusted-validator GdnSyH3YtwcxFvQrVVJMm1JhTS4QVX7MFsX56uJLUfiZ \
+    --trusted-validator DE1bawNcRJB9rVm3buyMVfr8mBEoyyu73NBovf2oXJsJ \
+    --trusted-validator CakcnaRDHka2gXyfbEd2d3xsvkJkqsLw2akB3zsN1D2S \
+    --expected-genesis-hash 5eykt4UsFv8P8NJdTREpY1vzqKqZKvdpKuc147dw2N9d \
+    --expected-shred-version 64864 \
+    # for your pain-less copy-paste
+
+# UPDATE docs/src/clusters.md TOO!!
+test_with_live_cluster "testnet" \
+    --entrypoint entrypoint.testnet.solana.com:8001 \
+    --trusted-validator 5D1fNXzvv5NjV1ysLjirC4WY92RNsVH18vjmcszZd8on \
+    --trusted-validator ta1Uvfb7W5BRPrdGnhP9RmeCGKzBySGM1hTE4rBRy6T \
+    --trusted-validator Ft5fbkqNa76vnsjYNwjDZUXoTWpP7VYm3mtsaQckQADN \
+    --trusted-validator 9QxCLckBiJc783jnMvXZubK4wH86Eqqvashtrwvcsgkv \
+    --expected-genesis-hash 4uhcVJyU9pJkvQyS88uRDiswHXSCkY3zQawwpjk2NsNY \
+    # for your pain-less copy-paste
+
```

Comment:
I wonder it's nice to have to upload fetched snapshots to the buildkite as artifacts for reproducible testing if anything odd happens.

Reply:
done by uploading snapshots only if the build failed.